From: Peter Bloomfield <peterbloomfield@bellsouth.net>
Date: Thu, 23 Jun 2022 21:30:51 +0000
Subject: In `test_datalist_id_remove_multiple()`,
 verify that the data list contains 3 items before calling
 `g_datalist_id_remove_multiple()`.

Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/2672
Origin: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2776
---
 glib/tests/dataset.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/glib/tests/dataset.c b/glib/tests/dataset.c
index 1d6feea..12eb33a 100644
--- a/glib/tests/dataset.c
+++ b/glib/tests/dataset.c
@@ -283,6 +283,10 @@ test_datalist_id_remove_multiple (void)
   g_datalist_id_set_data (&list, two, GINT_TO_POINTER (2));
   g_datalist_id_set_data (&list, three, GINT_TO_POINTER (3));
 
+  count = 0;
+  g_datalist_foreach (&list, foreach_func, &count);
+  g_assert_cmpint (count, ==, 3);
+
   g_datalist_id_remove_multiple (&list, keys, G_N_ELEMENTS (keys));
 
   count = 0;
